### DISCUSSION TITLE
Stats component does not show the total number of results in a multiple indexes search

### DISCUSSION THREAD

**Discussion by ghost** [algolia/instantsearch#6347] [Q&A] [answered]
Hi!

I'm using React InstantSearch to search multiple indexes in Algolia. 

I need to display the total number of results on top of my search page and I've understood that the Stats component is the way to go.

Unfortunately it only shows the results for the default index set in `indexName` on `InstantSearchNext` (indexA). Whenever I get a result from any of the other indexes (indexB or indexC) defined by the Index component, I get 0 results.

Is this as intended? Is there a workaround to show the total amount?

Many thanks! 

This is what I have:

```
<InstantSearchNext
	searchClient={searchClient}
	indexName="indexA"
	routing
>
        <SearchBox/>
	<Stats /> 
		<div>
			<Index indexName={indexA}>
				<Hits />
			</Index>

			<Index indexName={indexB}>
				<Hits />
			</Index>

			<Index indexName={indexC} >
				<Configure filters="_type:article" />
				<Hits />
			</Index>

			<Index indexName={indexC}>
				<Configure filters="NOT _type:article" />
				<Hits />
			</Index>
		</div>
</InstantSearchNext>
```

**Comment by Haroenv** [comment, marked as answer]
stats is only designed for a single index. A solution is using `useInstantSearch`: 

```js
function TotalStats() {
  const { scopedResults, results } = useInstantSearch();

  const totalNbHits = scopedResults.reduce((total, current) => total + current.results.nbHits, 0)

  return <p>found {totalNbHits} in {results.processingTimeMS}ms</p>
}
```

**Reply by ghost** [reply]
@Haroenv thank you! That was actually what I started looking at too. But do you know from where to import the ScopedResult[] type? 

<img width="431" alt="image" src="https://github.com/user-attachments/assets/3b0390a7-71ac-4394-b7d9-5c55e1f36bd0">

**Reply by Haroenv** [reply]
should be accessible from `import type { ScopedResult } from "instantsearch.js"` (also instead of using a set, then filtering, you probably can use a Map with indexName as the key and then get the values for rendering)

**Reply by ghost** [reply]
Im using react-instantsearch and not instantsearch.js. When trying to import the type from react-instantsearch it says it has no exported type for `ScopedResult`.

Is it a bad idea to. use set? My assumption was that set is quicker.

**Reply by Haroenv** [reply]
yes, React InstantSearch uses InstantSearch.js under the hood, so you can add it directly as a dependency if you don't want to rely on a transitive version, but make sure you update them in sync after that. I've had making sure types are re-exported on my todo list for a bit, but the Widget types are different so that could be confusing.

It's not a bad idea to use a set, I just mean you can create the set/map from the items directly instead of iterating twice.

**Reply by ghost** [reply]
I understand, worked to import the type like you suggested, thanks!

Just curious, what would the Map-version look like?

**Reply by Haroenv** [reply]
Thinking about it again, I'm not sure if it's actually more efficient if push comes to shove, but I had this in mind:

```
const deduped = [...new Map(scopedResults.map(res => [res.indexName, res]).values()]
```

**Reply by ghost** [reply]
I see. I don't know if that does what I'm after tho... For some reason the scopedResult sometimes returns an array with duplicate of indexes. So what I want is to remove those duplicates so that I dont get the wrong number back. What I'm doing now instead is:

```const removeDuplicates = (arr: ScopedResult[]) => {
			const map = new Map<string, boolean>()

			return arr.reduce((acc, item) => {
				if (!map.has(item.indexId)) {
					map.set(item.indexId, true)
					acc.push(item)
				}
				return acc
			}, [])
		}

**Reply by Haroenv** [reply]
doesn't my code already do that too, as `[...new Map([[1, 1], [1,2]]).values()]` returns `[2]`? In any case if your code works how you want it, go ahead :) I just think you're checking for duplicates twice

**Reply by ghost** [reply]
Maybe it does! I just looked at it briefly, my bad :) Thanks for all the help and feedback!
